### DISCUSSION TITLE
Why the Simulation.C in ThermalHydraulicModel can be the driver of the simulation?

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#24120] [Q&A Modules: General] [answered]
Hi all,

I'd like to know why the **_Simulation.C_** can be the driver of the simulation in THM? How does it work? It really confused me. Could anyone kindly tell me?

Thank you!
Best regards.

**Comment by GiudGiud** [comment]
@joshuahansel any doc page on Simulation?

**Reply by joshuahansel** [reply]
No there isn't. It wasn't prioritized since it would be more of a developer documentation.

**Comment by joshuahansel** [comment]
I don't think it's really necessary for your question, but note that `THMProblem` inherits from `FEProblem` and `Simulation`:
```
class THMProblem : public FEProblem, public Simulation
```

As far as it being a "driver", I'd say just look at the actions/tasks that are executing and let those be your guide. Some of these actions go into the simulation class to do various things. For example, the components are stored in the `Simulation` class, and there is an action to add a component. I think if we wanted to, if we could put all of `Simulation`'s data in a different place (like putting all of the components into MOOSE's existing warehouse), then we might be able to do away with `Simulation` and just have the "action" code directly in actions.

Let me know if that answers your question.

**Reply by SomnusYu** [reply]
Thank you for your reply! So the simulation is just to store data, is that right? I just like to find out how the THM works if we give an input file and run this input file. It should run line by line according to a main function. If the simulation is not the main function, which file is the main function?

**Reply by SomnusYu** [reply]
For example, if we use the FlowChannel1Phase component, which function decides on building mesh first or initializing first? It should have an underlying logic.

**Reply by joshuahansel** [reply, marked as answer]
The main function is at `modules/thermal_hydraulics/src/main.C`, but it's a bit difficult to follow it all the way down.

> which function decides on building mesh first or initializing first?

Actions are used to do those two tasks. Task dependencies with each other are registered, and a dependency resolver takes that information and sorts all of the tasks/actions. There is a basic documentation for components: https://mooseframework.inl.gov/syntax/Components/index.html that tells you the order that the various component methods will be executed in, if that's all you're interested in.

**Reply by SomnusYu** [reply]
OK, thank you very much!

**Reply by SomnusYu** [reply]
Hi joshuahansel,

Could you please tell me what is the difference between these two lines:
### syntax.registerActionSyntax("AddComponentAction", "Components/*", "THM:add_component");
### syntax.registerActionSyntax("AddComponentAction", "Components/*/\*", "THM:add_component");.


What's the meaning of this symbol: /* ? 
I also find some lines without this symbol, and it also doesn't have the task name, just like:
### syntax.registerActionSyntax("THMCreateMeshAction", "Components");

Thank you!

**Reply by GiudGiud** [reply]
This is for Regex. * means anything.

so Components/* means anything within the components block
Components/*/* means anything within any block inside the components block

**Reply by SomnusYu** [reply]
I got it, thank you!
